src/components/theme/ProductGrid.tsx

```diff
@@ -0,0 +1,72 @@
+import styled from "@emotion/styled";
+import ProductCard from "./ProductCard";
+import type { ProductInfo } from "@/types/product";
+
+interface Props {
+  products: ProductInfo[];
+  loader: React.RefObject<HTMLDivElement | null>;
+  loading: boolean;
+}
+
+export default function ProductGrid({ products, loader, loading }: Props) {
+  return (
+    <>
+      {products.length === 0 && loading ? (
+        <Spinner />
+      ) : products.length === 0 ? (
+        <EmptyBox>
+          <EmptyMessage>상품이 없습니다.</EmptyMessage>
+        </EmptyBox>
+      ) : (
+        <CardGrid>
+          {products.map((item) => (
+            <ProductCard key={item.id} item={item} />
+          ))}
+          <div ref={loader} style={{ height: 1 }} />
+        </CardGrid>
+      )}
```

Comment:
저는 보통 이렇게 삼항연산자가 복잡해지면, 즉시실행함수 + early return 형태를 사용합니다~!
사람마다 선호하는 형태가 달라서 알아두시기만 하셔도 좋을 것 같아요 ㅎㅎ

```suggestion
    {(() => {
        if (products.length === 0 && loading) {
          return <Spinner />;
        }

        if (products.length === 0) {
          return (
            <EmptyBox>
              <EmptyMessage>상품이 없습니다.</EmptyMessage>
            </EmptyBox>
          );
        }

        return (
          <CardGrid>
            {products.map((item) => (
              <ProductCard key={item.id} item={item} />
            ))}
            <div ref={loader} style={{ height: 1 }} />
          </CardGrid>
        );
      })()}
```